src/main/java/bio/terra/cda/app/controller/QueryApiController.java

```diff
@@ -95,12 +96,36 @@ private String createNextUrl(String jobId, int offset, int limit) {
   }
 
   @Override
-  public ResponseEntity<QueryCreatedData> uniqueValues(String version, String body) {
-
+  public ResponseEntity<QueryCreatedData> uniqueValues(String version, String body, String system) {
     String table = applicationConfiguration.getBqTable() + "." + version;
     NestedColumn nt = NestedColumn.generate(body);
-    String querySql = "SELECT DISTINCT " + nt.getColumn() + " FROM " + table + nt.getUnnestClause();
-    logger.debug("uniqueValues: " + querySql);
+    Set<String> unnestClauses = nt.getUnnestClauses();
+    StringBuffer unnestConcat = new StringBuffer();
+    String whereClause = null;
```

Comment:
Both arms of the `if` set `whereClause` so it can be declared `final` without an initial value. This is much better than setting it to `null`.
```suggestion
    final String whereClause;
```

Reply:
updated.